File: TabLayout/app/src/main/java/com/example/tablayout/MainActivity.java
package com.example.tablayout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;

import com.google.android.material.tabs.TabLayout;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TabLayout tab=findViewById(R.id.tab);
        ViewPager viewPager=findViewById(R.id.viewpager);


        ViewPagerMessengerAdapter adapter=new ViewPagerMessengerAdapter(getSupportFragmentManager());
        viewPager.setAdapter(adapter);
        tab.setupWithViewPager(viewPager);
    }
}

File: InstagramApp/app/src/main/java/com/example/instagramapp/Activity/MainActivity.kt
package com.example.instagramapp.Activity

import android.content.Intent
import android.graphics.Color
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.annotation.RequiresApi
import com.example.instagramapp.R
import com.google.firebase.auth.FirebaseAuth

class MainActivity : AppCompatActivity() {
    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        window.statusBarColor= Color.TRANSPARENT
        Handler(Looper.getMainLooper()).postDelayed({
           if (FirebaseAuth.getInstance().currentUser==null){
               startActivity(Intent(this, SignUpActivity::class.java))
               finish()
           } else{
               startActivity(Intent(this, HomeActivity::class.java))
               finish()
           }
        },3000)
    }
}